docs/settings.md

```diff
@@ -0,0 +1,120 @@
+# Zulip settings
+
+The page documents the Zulip settings system, and hopefully should
+help you decide how to correctly implement new settings you're adding
+to Zulip.  We have two types of administrative settings in Zulip:
+server settings (which are set via configuration files are apply to
+the whole Zulip installation), and realm settings (which are usually
+set via the /#administration page in the Zulip web application) and
+apply to a single Zulip realm/team (which for most Zulip servers is
+the only realm on the server).
+
+## Server settings
+
+Zulip uses the [Django settings
+system](https://docs.djangoproject.com/en/1.9/topics/settings/), which
+means that the settings files are Python programs that set a lot of
+variables with all-capital names like EMAIL_GATEWAY_PATTERN.  You can
+access these anywhere in the Zulip Django code using e.g.:
+
+```
+from django.conf import settings
+print(settings.EMAIL_GATEWAY_PATTERN)
+```
+
+Additionally, if you need to access a Django setting in a shell
+script (or just on the command line for debugging), you can use e.g.:
+
+```
+$ ./scripts/get-django-setting EMAIL_GATEWAY_PATTERN
+%s@localhost:9991
+```
+
+Zulip has separated those settings that we expect a system
+administrator to change (with nice documentation) from the ~1000 lines
+of settings needed by the Zulip Django app.  As a result, there are a
+few files involved in the Zulip settings for server administrations:
+
+* `/etc/zulip/settings.py` (generated from
+  `zproject/local_settings_template.py`) is the main system
+  administration facing settings file for Zulip.  It contains all the
+  server-specific settings, such as how to send outgoing email, the
+  hostname of the Postgres database, etc., but does not contain any
+  secrets (e.g. passwords, secret API keys, cryptographic keys, etc.).
+
+* `/etc/zulip/zulip-secrets.conf` (generated by
+  `scripts/setup/generate-secrets.py` as part of installation)
```

Comment:
as part of a production installation? (Would these exist in a dev installation?)